lib/proximity.js

```diff
@@ -448,41 +459,80 @@ function Proximity(opts) {
      */
     inches: {
       get: function() {
-        return +(this.centimeters * 0.39).toFixed(2);
+        return +(this.centimeters * 0.39).toFixed(3);
       }
     },
     in: {
       get: function() {
         return this.inches;
       }
     },
+    filtered: {
+      get: typeof opts.filter === "function" ? opts.filter : function() {
+        var average = this.samples.reduce(function(a, b) {
+          return a + b;
+        }) / this.samples.length;
+        var isSoften = typeof filterOpts.soften === "number";
+
+        raw = average;
+
+        return {
+          cm: isSoften ? soften(this.cm) : this.cm,
+          centimeters: isSoften ? soften(this.centimeters) : this.centimeters,
+          in : isSoften ? soften(this.in) : this.in,
+          inches: isSoften ? soften(this.inches) : this.inches,
+        };
+
+        function soften(num) {
+          return Math.round(num * Math.pow(10, filterOpts.soften)) / Math.pow(10, filterOpts.soften);
+        }
+      }
+    }
   });
 
   if (typeof this.initialize === "function") {
     opts.pinValue = pinValue;
     this.initialize(opts, function(data) {
+
+      if (isFiltered && typeof filterOpts.deviation === "number") {
+        data = raw && data > raw * (filterOpts.deviation) ? raw : data;
```

Comment:
I just realized something: I'm not sure this is the data that should be operated on because `data` could be _very_ different depending on the source peripheral. Shouldn't the filtering be applied to the appropriate data for the component? For example, the `Proximity` class computes centimeter values, so centimeter values should be what is filtered (instead of time in μs).